dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/credentials.js

```diff
@@ -14,6 +14,18 @@
 
 var Credentials = (function() {
 
+    var activeform = undefined;
+    var amazonform = Object.create(Forms.Form);
+    var userpwdform = Object.create(Forms.Form);
+    var cloudfoundryform = Object.create(Forms.Form);
+    var genericform = Object.create(Forms.Form);
+
+    var form_by_provider = {
+        "aws-ec2": amazonform,
+        "openshift": userpwdform,
+        "CloudFoundry": cloudfoundryform,
+    };
+
```

Comment:
It seems ok to me. @PaoloCifariello and @kiuby88 can provide more fruitful comments, as they are much more familiar than me about these "provider identifiers".